code/include/playfab_c/PlayFabClientDataModels_c.h

```diff
@@ -0,0 +1,495 @@
+    AE,
+    GB,
+    US,
+    UM,
+    UY,
+    UZ,
+    VU,
+    VE,
+    VN,
+    VG,
+    VI,
+    WF,
+    EH,
+    YE,
+    ZM,
+    ZW
+};
+
+enum class PlayFabSubscriptionProviderStatus : uint32_t
+{
+    NoError = 0,
+    Cancelled = 1,
+    UnknownError = 2,
+    BillingError = 3,
+    ProductUnavailable = 4,
+    CustomerDidNotAcceptPriceChange = 5,
+    FreeTrial = 6,
+    PaymentPending = 7
+};
+
+enum class PlayFabPushNotificationPlatform : uint32_t
+{
+    ApplePushNotificationService = 0,
+    GoogleCloudMessaging = 1
+};
+
+struct PlayFabPlayerProfileViewConstraints
+{
+    bool showAvatarUrl;
+    bool showBannedUntil;
+    bool showCampaignAttributions;
+    bool showContactEmailAddresses;
+    bool showCreated;
+    bool showDisplayName;
+    bool showExperimentVariants;
+    bool showLastLogin;
+    bool showLinkedAccounts;
+    bool showLocations;
+    bool showMemberships;
+    bool showOrigination;
+    bool showPushNotificationRegistrations;
+    bool showStatistics;
+    bool showTags;
+    bool showTotalValueToDateInUsd;
+    bool showValuesToDate;
+};
+
+struct PlayFabGetPlayerCombinedInfoRequestParams
+{
+    bool getCharacterInventories;
+    bool getCharacterList;
+    bool getPlayerProfile;
+    bool getPlayerStatistics;
+    bool getTitleData;
+    bool getUserAccountInfo;
+    bool getUserData;
+    bool getUserInventory;
+    bool getUserReadOnlyData;
+    bool getUserVirtualCurrency;
+    const char** playerStatisticNames;
+    size_t playerStatisticNamesCount;
+    PlayFabPlayerProfileViewConstraints* profileConstraints;
+    const char** titleDataKeys;
+    size_t titleDataKeysCount;
+    const char** userDataKeys;
+    size_t userDataKeysCount;
+    const char** userReadOnlyDataKeys;
+    size_t userReadOnlyDataKeysCount;
+};
+
+struct PlayFabLoginWithCustomIDRequest
+{
+    bool* createAccount;
+    const char* customId;
+    PlayFabDictionaryEntry* customTags;
+    size_t customTagsCount;
+    const char* encryptedRequest;
+    PlayFabGetPlayerCombinedInfoRequestParams* infoRequestParameters;
+    const char* playerSecret;
+    const char* titleId;
+};
+
+struct PlayFabAdCampaignAttributionModel
+{
+    time_t attributedAt;
+    const char* campaignId;
+    const char* platform;
+};
+
+struct PlayFabContactEmailInfoModel
+{
+    const char* emailAddress;
+    const char* name;
+    PlayFabEmailVerificationStatus* verificationStatus;
+};
+
+struct PlayFabLinkedPlatformAccountModel
+{
+    const char* email;
+    PlayFabLoginIdentityProvider* platform;
+    const char* platformUserId;
+    const char* username;
+};
+
+struct PlayFabLocationModel
+{
+    const char* city;
+    PlayFabContinentCode* pfContinentCode;
+    PlayFabCountryCode* pfCountryCode;
+    double* latitude;
+    double* longitude;
+};
+
+struct PlayFabSubscriptionModel
+{
+    time_t expiration;
+    time_t initialSubscriptionTime;
+    bool isActive;
+    PlayFabSubscriptionProviderStatus* status;
+    const char* subscriptionId;

```

Comment:
should be ptr?

Reply:
see above

---
In reply to: [590791909](https://github.com/jplafonta/XPlatCppSdk/pull/2#discussion_r590791909) [](ancestors = 590791909)